storage/bootstrap/bootstrapper/commitlog/source_prop_test.go

```diff
@@ -0,0 +1,235 @@
+package commitlog
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
+	"github.com/m3db/m3db/context"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/persist/fs/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+	"github.com/m3db/m3db/ts"
+	xtime "github.com/m3db/m3x/time"
+)
+
+func TestCommitLogSourcePropCorrectlyBootstrapsFromCommitlog(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.Rng.Seed(123456789)
+	parameters.MinSuccessfulTests = 100
+	props := gopter.NewProperties(parameters)
+
+	props.Property("Commitlog bootstrapping properly bootstraps the entire commitlog", prop.ForAll(
+		func(input propTestInput) (bool, error) {
+			// Make sure we have a new directory for each test run
+			dir, err := ioutil.TempDir("", "foo")
+			if err != nil {
+				return false, err
+			}
+
+			// Configure the commitlog to use the test directory and set the blocksize
+			commitLogOpts := commitlog.NewOptions().
+				SetBlockSize(2 * time.Hour).
+				SetFilesystemOptions(fs.NewOptions().SetFilePathPrefix(dir))
+			bootstrapOpts := testOptions().SetCommitLogOptions(commitLogOpts)
+
+			// Instantiate commitlog
+			log, err := commitlog.NewCommitLog(commitLogOpts)
+			if err != nil {
+				return false, err
+			}
+			err = log.Open()
+			if err != nil {
+				return false, err
+			}
+
+			// Write all the datapoints to the commitlog
+			for _, write := range input.writes {
+				log.Write(context.NewContext(), write.series, write.datapoint, write.unit, write.annotation)
+			}
+			err = log.Close()
+			if err != nil {
+				return false, err
+			}
+
+			// Instantiate a commitlog source
+			source := newCommitLogSource(bootstrapOpts)
```

Comment:
Could you use the exported ctor here instead?

Reply:
done